### PR TITLE
Add shards info command and make the song command prettier

### PR DESCRIPTION
**Pull request by binaryoverload**
- [x] Completed PR - Has all features or fixes that it was meant to and isn't half done
- [x] Your code follows the same code style as the rest of the project
- [x] Tested changes

## What is added
Added Apache Commons Lang
Added helper method to get shard ID
Added shard info command

Method to create progress bar
Better song info **Now with progress bar!**

### CODE REVIEW

File: src/main/java/stream/flarebot/flarebot/commands/secret/ShardInfoCommand.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import stream.flarebot.flarebot.util.MessageUtils;
+
+public class ShardInfoCommand implements Command {
+    @Override
+    public void onCommand(User sender, TextChannel channel, Message message, String[] args, Member member) {
+        StringBuilder builder = new StringBuilder(String.format("```diff\nTotal: %d\n", FlareBot.getInstance().getClients().length));
+        for (JDA jda : FlareBot.getInstance().getClients()) {
+            switch (jda.getStatus()) {
+                case CONNECTED:
+                    builder.append(String.format("+ Shard ID: %s - %s\n+     Guild Count: %d\n", MessageUtils.getShardId(jda), WordUtils.capitalizeFully(jda.getStatus().toString().replace("_", " ")), jda.getGuilds().size()));
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/commands/secret/ShardInfoCommand.java]: I said this before. Break strings after `\n`

File: src/main/java/stream/flarebot/flarebot/commands/secret/ShardInfoCommand.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                    break;
+                default:
+                    builder.append(String.format("- Shard ID: %s - %s\n+     Guild Count: %d\n", MessageUtils.getShardId(jda), WordUtils.capitalizeFully(jda.getStatus().toString().replace("_", " ")), jda.getGuilds().size()));
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/commands/secret/ShardInfoCommand.java]: Same

File: src/main/java/stream/flarebot/flarebot/util/ReportManager.java [hunk 3]
```diff
@@ -60,13 +59,14 @@ public Report getReport(String guildID, int id) {
...
+    public List<Report> getReportsToSave() {
         return reportsToSave;
     }
 
     private static ReportManager instance;
-    public static ReportManager getInstance(){
-        if(instance == null){
+
+    public static ReportManager getInstance() {
+        if (instance == null) {
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/util/ReportManager.java]: Again, you're making changes to irrelevant code. Please uncheck the reformat and organize imports options in the commit dialog. Please.

File: src/main/java/stream/flarebot/flarebot/commands/secret/ShardInfoCommand.java (hunk 1 above)
Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/secret/ShardInfoCommand.java]: Oops yeah

**Comment by binaryoverload** [comment]
Thank god <3

**Comment by WalshyDev** [comment]
Before I approve this I think we can use this to improve the ascii table method a bit. It might be a good idea to add another param to it with the Lang and obviously if null is passed just leave it as a no Lang code block.

File: src/main/java/stream/flarebot/flarebot/commands/general/ShardInfoCommand.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import stream.flarebot.flarebot.FlareBot;
+import stream.flarebot.flarebot.commands.Command;
+import stream.flarebot.flarebot.commands.CommandType;
+import stream.flarebot.flarebot.util.MessageUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ShardInfoCommand implements Command {
+    @Override
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/commands/general/ShardInfoCommand.java]: Space here.

File: src/main/java/stream/flarebot/flarebot/commands/general/ShardInfoCommand.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    public void onCommand(User sender, TextChannel channel, Message message, String[] args, Member member) {
+        channel.sendMessage(buildTable(FlareBot.getInstance().getClients())).queue();
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/commands/general/ShardInfoCommand.java]: I don't see why you're making a method just to call it the one time here.

File: src/main/java/stream/flarebot/flarebot/commands/general/ShardInfoCommand.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    public String getUsage() {
+        return "{%}shardsinfo";
+    }
+
+    @Override
+    public CommandType getType() {
+        return CommandType.GENERAL;
+    }
+
+    public String buildTable(JDA[] jdas) {
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/commands/general/ShardInfoCommand.java]: If you are gonna make it a method (No reason to) make this private as it isn't used anywhere else.

File: src/main/java/stream/flarebot/flarebot/commands/music/SongCommand.java [hunk 7]
```diff
@@ -66,12 +66,23 @@ public CommandType getType() {
...
                 + (minutes < 10 ? "0" + minutes : minutes) + ":" + (seconds < 10 ? "0" + seconds : seconds);
     }
+
+    public static String getProgressBar(Track track) {
+        float percentage = (100f / track.getTrack().getDuration() * track.getTrack().getPosition());
+        StringBuilder progress = new StringBuilder("[");
+        progress.append(StringUtils.repeat("▬", (int) Math.round((double) percentage / 10)));
+        progress.append("]()");
+        progress.append(StringUtils.repeat("▬", 10 - (int) Math.round((double) percentage / 10)));
+        progress.append(" " + new DecimalFormat("#.##").format(percentage) + "%");
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/commands/music/SongCommand.java]: Don't make a new DecimalFormat every time

File: src/main/java/stream/flarebot/flarebot/util/MessageUtils.java [hunk 8]
```diff
@@ -230,7 +236,7 @@ public static String makeAsciiTable(java.util.List<String> headers, java.util.Li
                 }
             }
         }
-        sb.append("```").append("\n");
+        sb.append("```" + lang).append("\n");
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/util/MessageUtils.java]: Shouldn't concat in an append. Instead that should be sb.append("```").append(lang).append("\n)

**Comment by binaryoverload** [comment]
:+1:

**Pull request merged**